model-api/xml-model/src/main/java/org/camunda/bpm/model/xml/impl/validation/ModelValidationResultsImpl.java

```diff
@@ -75,9 +74,50 @@ public void write(StringWriter writer, ValidationResultFormatter formatter) {
     }
   }
 
+  @Override
+  public void write(StringWriter writer, ValidationResultFormatter formatter, int maxSize) {
+    int printedCount = 0;
+    int previousLength = 0;
+    for (var entry : collectedResults.entrySet()) {
+      var element = entry.getKey();
+      var results = entry.getValue();
+
+      formatter.formatElement(writer, element);
+
+      for (var result : results) {
+        formatter.formatResult(writer, result);
+
+        // Size and Length are not necessarily the same, depending on the encoding of the string.
+        int currentSize = writer.getBuffer().toString().getBytes().length;
+        int currentLength = writer.getBuffer().length();
+        if (!canAccommodateResult(maxSize, currentSize, printedCount)) {
+          writer.getBuffer().setLength(previousLength);
```

Comment:
👍 Nice trick to deal with the problem that you don't know how many bytes will be written before formatting. Also good that you've described the general approach in the commit message.